src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/error/HttpResponseErrorMessageMapper.java

```diff
@@ -0,0 +1,56 @@
+package io.stargate.sgv2.jsonapi.service.embedding.operation.error;
+
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import jakarta.ws.rs.core.Response;
+
+public class HttpResponseErrorMessageMapper {
+  /**
+   * This method returns default exception based on the http response. Individual provider can
+   * provide an override
+   *
+   * @param response
+   * @return
+   */
+  public static RuntimeException getDefaultException(Response response) {
+
+    // Status code == 408 and 504 for timeout
+    if (response.getStatus() == Response.Status.REQUEST_TIMEOUT.getStatusCode()
+        || response.getStatus() == Response.Status.GATEWAY_TIMEOUT.getStatusCode()) {
+      final JsonApiException apiException = ErrorCode.EMBEDDING_PROVIDER_TIMEOUT.toApiException();
+      return new RuntimeException(apiException);
```

Comment:
Since `JsonApiException` is a `RuntimeException`, is there need to add this wrapping?
(I guess this also answers my earlier question about wrapping)